packages/server-side-render/src/server-side-render.js

```diff
@@ -20,126 +20,105 @@ export function rendererPath( block, attributes = null, urlQueryArgs = {} ) {
 	} );
 }
 
-export class ServerSideRender extends Component {
-	constructor( props ) {
-		super( props );
-		this.state = {
-			response: null,
-		};
-	}
+export function ServerSideRender( props ) {
+	const {
+		className,
+		EmptyResponsePlaceholder,
+		ErrorResponsePlaceholder,
+		LoadingResponsePlaceholder,
+	} = props;
 
-	componentDidMount() {
-		this.isStillMounted = true;
-		this.fetch( this.props );
-		// Only debounce once the initial fetch occurs to ensure that the first
-		// renders show data as soon as possible.
-		this.fetch = debounce( this.fetch, 500 );
-	}
+	const [ response, setResponse ] = useState( null );
+	const [ isLoading, setIsLoading ] = useState( false );
+	const isMounted = useRef( true );
 
-	componentWillUnmount() {
-		this.isStillMounted = false;
-	}
+	useEffect( () => {
+		fetchData( props );
 
-	componentDidUpdate( prevProps ) {
-		if ( ! isEqual( prevProps, this.props ) ) {
-			this.fetch( this.props );
-		}
-	}
+		fetchData = debounce( fetchData, 500 );
+		return () => {
+			isMounted.current = false;
+		};
+	}, [] );
 
-	fetch( props ) {
-		if ( ! this.isStillMounted ) {
+	useEffect( () => {
+		fetchData( props );
+	}, [ props ] );
+
+	function fetchData( options ) {
```

Comment:
Passing in options here isn't really necessary, because this function is already within the scope of the parent component, which means we can just use the props directly.

Reply:
Agreed and done  👍